pennylane/decomposition/decomposition_graph.py

```diff
@@ -162,10 +164,30 @@ def _get_decompositions(self, op_node: CompressedResourceOp) -> list[Decompositi
 
         decomps = self._alt_decomps.get(op_name, []) + list_decomps(op_name)
 
-        if issubclass(op_node.op_type, qml.ops.Adjoint):
+        if (
+            issubclass(op_node.op_type, qml.ops.Adjoint)
+            and self_adjoint not in decomps
+            and adjoint_rotation not in decomps
+        ):
+            # In general, we decompose the adjoint of an operator by applying adjoint to the
+            # decompositions of the operator. However, this is not necessary if the operator
+            # is self-adjoint or if it has a single rotation angle which can be trivially
+            # inverted to obtain its adjoint. In this case, `self_adjoint` or `adjoint_rotation`
+            # would've already been retrieved as a potential decomposition rule for this
+            # operator, so there is no need to consider the general case.
             decomps.extend(self._get_adjoint_decompositions(op_node))
 
-        elif issubclass(op_node.op_type, qml.ops.Pow):
+        elif (
+            issubclass(op_node.op_type, qml.ops.Pow)
+            and pow_rotation not in decomps
+            and pow_involutory not in decomps
+        ):
+            # Similar to the adjoint case, the `_get_pow_decompositions` contains the general
+            # approach we take to decompose powers of operators. However, if the operator is
+            # involutory or if it has a single rotation angle that can be trivially multiplied
+            # with the power, we would've already had retrieved `pow_involutory` or `pow_rotation`
```

Comment:
```suggestion
            # with the power, we would've already retrieved `pow_involutory` or `pow_rotation`
```

Grammarly flags this as an error, but maybe it is correct